cedar-policy-core/src/authorizer.rs

```diff
@@ -165,16 +165,7 @@ impl Authorizer {
         pset: &PolicySet,
         entities: &Entities,
     ) -> ResponseKind {
-        let eval = match Evaluator::new(q, entities, &self.extensions) {
-            Ok(eval) => eval,
-            Err(e) => {
-                return ResponseKind::FullyEvaluated(Response::new(
-                    Decision::Deny,
-                    HashSet::new(),
-                    vec![AuthorizationError::AttributeEvaluationError(e)],
-                ));
-            }
-        };
+        let eval = Evaluator::new(q.clone(), entities, &self.extensions);
```

Comment:
`Request` is probably cheap to clone, but would it still be better to take an owned `Request` here?

Reply:
That's fair, for Core here.  I don't think it's worth changing (breaking) the public API to take owned `Request`, especially because maybe in the future we find a way to avoid this clone.

Before this PR, the `Request` was effectively cloned in `Evaluator::new()`.  I've just made it more explicit by having `Evaluator::new()` take owned `Request` itself.  The end result in terms of performance should be the same before and after this PR (in this respect)

Reply:
Refactored in Core, left the clone in api.rs